webapp/main.py

```diff
@@ -1,30 +1,120 @@
 from flask import Flask, redirect, url_for, request, render_template
-from pytube import extract
+from pytube import extract, YouTube
+import os, whisper, json
+import datetime
+import pathlib
 
 app = Flask(__name__)
  
 @app.route('/', methods=['GET'])
 def home():
-    return render_template('form.html')
+    file = open('video_list.txt',mode='r')
+    video_list = file.read()
+    # close the file
+    file.close()
+    return render_template('form.html', video_list=video_list)
  
+
 @app.route('/success/<name>/<lan>')
 def success(name, lan):
     return render_template('index.html', name=name, lan=lan)
+
+
+def timedelta_to_videotime(delta):
+    """
+    Here's a janky way to format a 
+    datetime.timedelta to match 
+    the format of vtt timecodes. 
+    """
+    parts = delta.split(":")
+    if len(parts[0]) == 1:
+        parts[0] = f"0{parts[0]}"
+    new_data = ":".join(parts)
+    parts2 = new_data.split(".")
+    if len(parts2) == 1:
+        parts2.append("000")
+    elif len(parts2) == 2:
+        parts2[1] = parts2[1][:3]
+    final_data = ".".join(parts2)
+    return final_data
+
+
+def whisper_segments_to_vtt_data(result_segments):
+  """
+  This function iterates through all whisper
+  segements to format them into WebVTT.
+  """
+  data = "WEBVTT\n\n"
+  for idx, segment in enumerate(result_segments):
+    num = idx + 1
+    data+= f"{num}\n"
+    start_ = datetime.timedelta(seconds=segment.get('start'))
+    start_ = timedelta_to_videotime(str(start_))
+    end_ = datetime.timedelta(seconds=segment.get('end'))
+    end_ = timedelta_to_videotime(str(end_))
+    data += f"{start_} --> {end_}\n"
+    text = segment.get('text').strip()
+    data += f"{text}\n\n"
+  return data
+
+
+def transcribe_from_link(link):
+	if link != "":
+		print("Transcribing...")	
+		yt = YouTube(link)
+		video = yt.streams.filter(only_audio=True).first()
+		
+		# # check for destination to save file
+		destination = 'static\\resources'
+
+		id = extract.video_id(link)
+		
+		# download the file
+		out_file = yt.streams.filter(progressive = True, file_extension = "mp4").first().download(output_path=destination, filename = id + ".mp4")

```

Comment:
Were we not supposed to use .mp3 to transcribe? Is it because we need to have .mp4 for the embedded video?

Reply:
Yeah, until we know how to run embeded YouTube video with the transcript, we use mp4 to render the video for now.